add oklch support

mohammed-r9 opened the pull request:
since oklch became so popular especially with shadcn/ui, it is feels necessary to support it

mohammed-r9 committed:
add oklch support

davychhouk commented:
This is a good one, `oklch` support is necessary now.

mohammed-r9 commented:
> This is a good one, `oklch` support is necessary now.

Surely, but the maintainers don't seem to see the PR or something. You can use my fork (as I'm doing currently) if you want

davychhouk commented:
This is working well, thanks.

<img width="520" height="204" alt="image" src="https://github.com/user-attachments/assets/b19e8e30-f1d8-4aba-9c0c-fa9854a5cc2f" />

mohammed-r9 commented:
> This is working well, thanks.
> 
> <img width="520" height="204" alt="image" src="https://github.com/user-attachments/assets/b19e8e30-f1d8-4aba-9c0c-fa9854a5cc2f" />
> 

Happy to hear that!

CemDK committed:
Add OKLCH color support

Implements OKLCH color format support as proposed in #169 by
@neckbeard-69.

  This PR includes only the functional changes from #169 without
formatting
  modifications, making it easier to review. All OKLCH conversion logic
and
  test cases are based on @neckbeard-69's work.

I am not going to open a PR, @neckbeard-69 please feel free to not
include the formatting changes, as I am sure that might make reviewing
this PR easier

brenoprata10 commented:
Could you rebase this PR, will review it today :)

mohammed-r9 commented:
> Could you rebase this PR, will review it today :)

done

brenoprata10 commented:
Could you also remove all of the formatting? Ideally, keep only the modified lines in your PR :)

mohammed-r9 commented:
> Could you also remove all of the formatting? Ideally, keep only the modified lines in your PR :)

could you please provide your `stylua` config?

davychhouk commented:
Hi @mohammed-r9,

I'm having this error on using your branch.

```
Error executing lua callback: ...vim-highlight-colors/lua/nvim-highlight-colors/utils.lua:55: bad argument #2 to 'get_color_value' (number expected, got nil)
stack traceback:
	[C]: in function 'get_color_value'
	...vim-highlight-colors/lua/nvim-highlight-colors/utils.lua:55: in function 'create_highlight'
	...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:137: in function 'highlight_colors'
	...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:169: in function 'refresh_highlights'
	...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:282: in function <...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:280>
	[C]: in function 'nvim_exec_autocmds'
	.../neovim/0.11.6/share/nvim/runtime/lua/vim/lsp/client.lua:1037: in function 'on_attach'
	.../neovim/0.11.6/share/nvim/runtime/lua/vim/lsp/client.lua:575: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
```

Please have a look, thanks.

mohammed-r9 commented:
> Hi @mohammed-r9,
> 
> I'm having this error on using your branch.
> 
> ```
> Error executing lua callback: ...vim-highlight-colors/lua/nvim-highlight-colors/utils.lua:55: bad argument #2 to 'get_color_value' (number expected, got nil)
> stack traceback:
> 	[C]: in function 'get_color_value'
> 	...vim-highlight-colors/lua/nvim-highlight-colors/utils.lua:55: in function 'create_highlight'
> 	...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:137: in function 'highlight_colors'
> 	...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:169: in function 'refresh_highlights'
> 	...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:282: in function <...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:280>
> 	[C]: in function 'nvim_exec_autocmds'
> 	.../neovim/0.11.6/share/nvim/runtime/lua/vim/lsp/client.lua:1037: in function 'on_attach'
> 	.../neovim/0.11.6/share/nvim/runtime/lua/vim/lsp/client.lua:575: in function ''
> 	vim/_editor.lua: in function <vim/_editor.lua:0>
> ```
> 
> Please have a look, thanks.


Hi, the feature is available in css lsp by default, I don't think it's necessary to use this plugin anymore to be honest

davychhouk commented:
> > Hi @mohammed-r9,
> > I'm having this error on using your branch.
> > ```
> > Error executing lua callback: ...vim-highlight-colors/lua/nvim-highlight-colors/utils.lua:55: bad argument #2 to 'get_color_value' (number expected, got nil)
> > stack traceback:
> > 	[C]: in function 'get_color_value'
> > 	...vim-highlight-colors/lua/nvim-highlight-colors/utils.lua:55: in function 'create_highlight'
> > 	...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:137: in function 'highlight_colors'
> > 	...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:169: in function 'refresh_highlights'
> > 	...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:282: in function <...nvim-highlight-colors/lua/nvim-highlight-colors/init.lua:280>
> > 	[C]: in function 'nvim_exec_autocmds'
> > 	.../neovim/0.11.6/share/nvim/runtime/lua/vim/lsp/client.lua:1037: in function 'on_attach'
> > 	.../neovim/0.11.6/share/nvim/runtime/lua/vim/lsp/client.lua:575: in function ''
> > 	vim/_editor.lua: in function <vim/_editor.lua:0>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Please have a look, thanks.
> 
> Hi, the feature is available in css lsp by default, I don't think it's necessary to use this plugin anymore to be honest

You're right, all good for now, thanks.
